### ISSUE TITLE
Maven JS obfuscated file contains wrong code

### ISSUE THREAD

**Issue by sseide** [eclipse/paho.mqtt.javascript#56]
The mvn compress goal produces wrong code.
The regular Message.prototype contains the getPayloadString() and getPayloadBytes() functions.
After obfuscation this functions are renamed to "getPayloadStringfunction" and "getPayloadBytesfunction". 
The ClientImpl prototype method store() is not updated accordingly and refers in the obfuscated version to the original getPayloadBytes - therefor all calls to send() with a qos=1 message fail for the mqttws31-min.js file.

I have not checked for more errors. At least current minified version is unusable.

Original file:

```
 ClientImpl.prototype.store = function(prefix, wireMessage) {
		var storedMessage = {type:wireMessage.type,  messageIdentifier:wireMessage.messageIdentifier, version:1};
	
		switch(wireMessage.type) {
		  case MESSAGE_TYPE.PUBLISH:
			  if(wireMessage.pubRecReceived)
				  storedMessage.pubRecReceived = true;
			  
			  // Convert the payload to a hex string.
			  storedMessage.payloadMessage = {};
			  var hex = "";
			  var messageBytes = wireMessage.payloadMessage.payloadBytes;
			  for (var i=0; i<messageBytes.length; i++) {
				if (messageBytes[i] <= 0xF)

...

	Message.prototype = {
		get payloadString() { return this._getPayloadString(); },
		get payloadBytes() { return this._getPayloadBytes(); },
		...

```

Minified/compressed version (with code reformatted for easier reading):

```
    v.prototype.store = function (E, D) {
        var A = {type: D.type, messageIdentifier: D.messageIdentifier, version: 1};
        switch (D.type) {
            case o.PUBLISH:
                if (D.pubRecReceived) {
                    A.pubRecReceived = true
                }
                A.payloadMessage = {};
                var C = "";
                var B = D.payloadMessage.payloadBytes;
                for (var z = 0; z < B.length; z++) {
                    if (B[z] <= 15) {

...

    a.prototype = {
        get payloadStringfunction() {
            return this._getPayloadString()
        }, get payloadBytesfunction() {
            return this._getPayloadBytes()
        }, get destinationNamefunction() {
        ...

```

Testcalls:

```
var message = new Paho.MQTT.Message(1);
message.destinationName = 'topic';
message.qos = 1;
mqttClient.send(message);

```

**Comment by sseide** [comment]
found closed issue #14  - if this issue is closed with "we do not use our own minified version, we use some other minifier instead" please, at lease, disable the corresponding maven goal as it still produces wrong code an let people waste lots of time tracking down this already known issue...

**Comment by jpwsutton** [comment]
hi @sseide, Sorry that you had to find the fix that way. I've now fixed this bug properly by switching to the closure compiler instead. This fix is now in the develop & master branch.
